src/restruct_poc/beams.hpp

```diff
@@ -0,0 +1,609 @@
+    oturb::View_Nx6 node_u_dot;           // State: translation/rotation velocity
+    oturb::View_Nx6 node_u_ddot;          // State: translation/rotation acceleration
+    oturb::View_Nx6 node_force_app;       // Applied forces
+    oturb::View_Nx6 node_force_elastic;   // Elastic forces
+    oturb::View_Nx6 node_force_gravity;   // Gravity forces
+    oturb::View_Nx6 node_force_inertial;  // Inertial forces
+    oturb::View_Nx6 node_force_external;  // External forces
+    oturb::View_Nx6 node_residual;        // Residual forces
+    oturb::View_NxN node_iteration;       // Iteration matrix
+
+    // Quadrature point data
+    oturb::View_N qp_weight;       // Integration weights
+    oturb::View_N qp_jacobian;     // Jacobian vector
+    oturb::View_Nx6x6 qp_Mstar;    // Mass matrix in material frame
+    oturb::View_Nx6x6 qp_Cstar;    // Stiffness matrix in material frame
+    oturb::View_Nx3 qp_x0;         // Initial position
+    oturb::View_Nx3 qp_x0_prime;   // Initial position derivative
+    oturb::View_Nx4 qp_r0;         // Initial rotation
+    oturb::View_Nx3 qp_u;          // State: translation displacement
+    oturb::View_Nx3 qp_u_prime;    // State: translation displacement derivative
+    oturb::View_Nx3 qp_u_dot;      // State: translation velocity
+    oturb::View_Nx3 qp_u_ddot;     // State: translation acceleration
+    oturb::View_Nx4 qp_r;          // State: rotation
+    oturb::View_Nx4 qp_r_prime;    // State: rotation derivative
+    oturb::View_Nx3 qp_omega;      // State: angular velocity
+    oturb::View_Nx3 qp_omega_dot;  // State: position/rotation
+    oturb::View_Nx6 qp_strain;     //
+    oturb::View_Nx6 qp_Fc;         //
+    oturb::View_Nx6 qp_Fd;         //
+    oturb::View_Nx6 qp_Fi;         //
+    oturb::View_Nx6 qp_Fg;         //
+    oturb::View_Nx6 qp_Fext;       //
+    oturb::View_Nx6x6 qp_RR0;      //
+    oturb::View_Nx6x6 qp_Muu;      //
+    oturb::View_Nx6x6 qp_Cuu;      //
+    oturb::View_Nx6x6 qp_Ouu;      //
+    oturb::View_Nx6x6 qp_Puu;      //
+    oturb::View_Nx6x6 qp_Quu;      //
+    oturb::View_Nx6x6 qp_Guu;      //
+    oturb::View_Nx6x6 qp_Kuu;      //
+
+    oturb::View_NxN shape_interp;  // shape function matrix for interpolation [Nodes x QPs]
+    oturb::View_NxN shape_deriv;   // shape function matrix for derivative interp [Nodes x QPs]
+
+    // Scratch variables to be replaced later
+    oturb::View_Nx6x6 M_6x6;   //
+    oturb::View_Nx3x4 M_3x4;   //
+    oturb::View_Nx3x3 M1_3x3;  //
+    oturb::View_Nx3x3 M2_3x3;  //
+    oturb::View_Nx3x3 M3_3x3;  //
+    oturb::View_Nx3x3 M4_3x3;  //
+    oturb::View_Nx3x3 M5_3x3;  //
+    oturb::View_Nx3x3 M6_3x3;  //
+    oturb::View_Nx3x3 M7_3x3;  //
+    oturb::View_Nx3x3 M8_3x3;  //
+    oturb::View_Nx3x3 M9_3x3;  //
+    oturb::View_Nx3 V1_3;      //
+    oturb::View_Nx3 V2_3;      //
+    oturb::View_Nx3 V3_3;      //
+    oturb::View_Nx4 qp_quat;   //
+
+    Beams(const BeamsInit beams_init)
```

Comment:
This feels like it should be either a factory free function or a method on BeamsInit.  BeamsInit can be coupled to Beams, but Beams should not be coupled to BeamsInit, if that makes sense.

Reply:
Yeah, I'm not really happy with the way that is organized either. I have it that way because it was easy to create and initialize the `Beams` type in the test fixture. I'm open to alternatives (I'm not a C++ expert by any means and I'm probably not doing it in the most appropriate way).